v4/newrelic/transaction.go

```diff
@@ -204,7 +210,16 @@ func (txn *Transaction) StartSegment(name string) *Segment {
 //
 // StartExternalSegment calls InsertDistributedTraceHeaders, so you don't need
 // to use it for outbound HTTP calls: Just use StartExternalSegment!
-func (txn *Transaction) InsertDistributedTraceHeaders(hdrs http.Header) {}
+func (txn *Transaction) InsertDistributedTraceHeaders(hdrs http.Header) {
+	if txn == nil {
+		return
+	}
+
+	txn.thread.currentSpan.Lock()
+	defer txn.thread.currentSpan.Unlock()
+
+	propagation.InjectHTTP(txn.thread.currentSpan.ctx, txn.app.propagators, hdrs)
```

Comment:
I think I have a method already called `getCurrentSpan` that does the locking and everything 💜 🟣 🟪

Reply:
Also, I believe this line will panic in this case:

```go
txn := &Transaction{}
txn.InsertDistributedTraceHeaders(hdrs)
```

Reply:
True. Fixed and added a test.